npm/webpack-dev-server-fresh/src/helpers/nextHandler.ts

```diff
@@ -0,0 +1,157 @@
+import type { CreateFinalWebpackConfig } from '../createWebpackDevServer'
+import debugLib from 'debug'
+import type { Configuration } from 'webpack'
+import * as fs from 'fs'
+import * as path from 'path'
+
+type PresetHandler = Omit<CreateFinalWebpackConfig, 'frameworkConfig'>
+
+const debug = debugLib('cypress:webpack-dev-server-fresh:nextHandler')
+
+export async function nextHandler ({ devServerConfig, sourceWebpackModulesResult }: PresetHandler) {
+  const webpackConfig = await loadWebpackConfig({ devServerConfig, sourceWebpackModulesResult })
+
+  debug('resolved next.js webpack config %o', webpackConfig)
+
+  checkSWC(webpackConfig, devServerConfig.cypressConfig)
+
+  // Next webpack compiler ignored watching any node_modules changes, but we need to watch
+  // for changes to 'dist/browser.js' in order to detect new specs that have been added
+  if (webpackConfig.watchOptions && Array.isArray(webpackConfig.watchOptions.ignored)) {
+    webpackConfig.watchOptions = {
+      ...webpackConfig.watchOptions,
+      ignored: [...webpackConfig.watchOptions.ignored.filter((pattern: string) => !/node_modules/.test(pattern)), '**/node_modules/!(@cypress/webpack-dev-server/dist/browser.js)**'],
+    }
+
+    debug('found options next.js watchOptions.ignored %O', webpackConfig.watchOptions.ignored)
+  }
+
+  return webpackConfig
+}
+
+/**
+ * Acquire the modules needed to load the Next webpack config
```

Comment:
I'm not really following this comment. Acquire them how, for what, why, from where?

Reply:
Addressed in ac29f02aa37dc2f16e412f546093b9f48e0cf710